src/vmspawn/vmspawn.c

```diff
@@ -1002,9 +1040,20 @@ static int run(int argc, char *argv[]) {
         if (r < 0)
                 goto finish;
 
+        r = sd_listen_fds_with_names(true, &names);
+        if (r < 0)
+                goto finish;
+
+        for (size_t i = 0; i < strv_length(names); i++) {
+                if (streq(names[i], "kvm"))
+                    kvm_device_fd = SD_LISTEN_FDS_START + i;
+                if (streq(names[i], "vhost-vsock"))
+                    vhost_device_fd = SD_LISTEN_FDS_START + i;
```

Comment:
it would make sense to print a log message about fds that were passed in that we cannot make sense of, and close them (the log message should make it clear that they are closed now)

Reply:
Fixed in 56f995436f.